cpp/src/arrow/gpu/cuda_test.cc

```diff
@@ -213,6 +215,67 @@ TEST_F(TestCudaDevice, Copy) {
   }
 }
 
+TEST_F(TestCudaDevice, CreateSyncEvent) {
```

Comment:
Doesn't have to be this test but it would be good to destroy the sync event and then check the cuda error code to have an explicit test that the destructor works as expected as well

Reply:
@kkraus14 What function should I call and check the error code of? Using `cuEventQuery` on the already deleted event has a segfault rather than returning an error I can check. Is there a different function that would do what you're suggesting but not segfault?

Reply:
I think I wrote this comment before the changes making the SyncEvents have RAII semantics. This is good now.